### PR TITLE
hasVisibleClients issue at chrome extension with manifest v3

### PR DESCRIPTION
**Pull request by embbnux**
hasVisibleClients can't work at chrome extension with manifest v3. 
Issue can be reproduced with [this project](https://github.com/embbnux/chrome-extension-manifest-v3-with-fcm/tree/firebase-issue-at-v3)

### CODE REVIEW

**Review by zwu52 — APPROVED** [review]
LGTM. thanks for the PR

File: packages/messaging/src/listeners/sw-listeners.ts [hunk 1]
```diff
@@ -207,11 +207,29 @@ async function getWindowClient(url: URL): Promise<WindowClient | null> {
   return null;
 }
 
+/**
+ * @return If runtime is chrome extension manifest v3.
```
Comment (egilmorez) [inline packages/messaging/src/listeners/sw-listeners.ts:211]: Should this be something more like, "If runtime is Chrome, use extension manifest v3?"

I feel like it needs a verb (especially if this is a public string parsed into reference docs).

**Review by egilmorez — COMMENTED** [review]
Thanks for sending!

All looks good but I do have one suggestion on a comment string.

File: packages/messaging/src/listeners/sw-listeners.ts (hunk 1 above)
Comment (zwu52) [reply packages/messaging/src/listeners/sw-listeners.ts]: sg. updated.

**Review by egilmorez — APPROVED** [review]
LG, thanks!

**Comment by zwu52** [comment]
hmm. Looks like `chrome` object isn't an object of the `ServiceWorkerGlobalScope`.  What's the error thrown when running chrome extension with manifest v3?

**Comment by embbnux** [comment]
> hmm. Looks like `chrome` object isn't an object of the `ServiceWorkerGlobalScope`. What's the error thrown when running chrome extension with manifest v3?

`chrome` is an object of `ServiceWorkerGlobalScope` at extension manifest v3:

image

For this PR, it fix `hasVisibleClients` issue at chrome extension manifest v3. It always be false for chrome extension manifest v3.

**Pull request closed without merging**